pause (Frame Advance) OK

hensldm opened the pull request:
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---
<!-- Leave the text above intact. Add additional comments below. -->
Matched z_pause.c, which turned out to be FrameAdvance init and update, so renamed file to z_frame_advance.c

Kelebek1 commented on src/code/z_frame_advance.c:
```diff
@@ -0,0 +1,25 @@
+
+/**
+ * Frame advance allows you to advance through the game one frame at a time on command.
+ * To advance a frame, hold Z and press R on the spceified controller.
```
specified

AngheloAlf reviewed (changes requested):
Could you rename this file to `z_frame_advance` in `tools/disasm/files.txt` too?

engineer124 reviewed (approved):
lgtm

EllipticEllipsis commented on src/code/z_frame_advance.c:
```diff
@@ -0,0 +1,25 @@
+    frameAdvCtx->enabled = false;
+}
+
+/**
+ * Frame advance allows you to advance through the game one frame at a time on command.
+ * To advance a frame, hold Z and press R on the specified controller.
+ * Holding Z and R will advance a frame every half second.
+ *
+ * This function returns true when frame advance is not active (game will run normally)
+ *
+ * Note: There is no way to enable it
+ */
```
Since this comment really applies to the whole system, it seems better to put it at the top of the file in a file header.

EllipticEllipsis reviewed (changes requested):
Are these functions completely unused? I'd explain what is required to be able to use this somewhere: you can check how OoT has it set up, for example.

On a more general note, I don't like renaming this file from what is far more likely to be a correct name: that these are in alphabetical order is very useful information for understanding how `code` is laid out and how we can expect it to be laid out in other versions, and renaming it loses that information. I seriously doubt that anyone would expect more than a very rough idea of what a file does from its name anyway: that's the whole point of having top-of-file descriptions.

(Note that my view on this is specific to files and to doing it *now*: I am all in favour of renaming functions etc, but I do not think this is a good time to start renaming files that have a file name that reflects their position like this.)

Kenix3 reviewed (commented):
I agree with Elliptic on this case. It's too early to rename files and knowing the order can be helpful. I think we should set the name back to z_pause.

hensldm commented:
Ok, reverted the file name change for now (though added a note on the sheet that this is frame advance). Also moved the comment section as a header, and added a note of what code could possibly be added to enable it.

EllipticEllipsis commented on src/code/z_pause.c:
```diff
@@ -1,19 +1,29 @@
+/**
```
I think I'd like to start this off with `File:` and `Description:` lines like in overlay files... we do need to discuss how we'd like to format them (should the initial description be a synopsis and then a fuller description given under a different header, for example?

hensldm replied:
Added `File:` and `Description:` lines to the header.
